packages/adp-tooling/src/client/abap-provider/index.ts

```diff
@@ -0,0 +1,124 @@
+import { isAppStudio } from '@sap-ux/btp-utils';
+import type { AbapTarget } from '@sap-ux/ui5-config';
+import type { Logger, ToolsLogger } from '@sap-ux/logger';
+import { createAbapServiceProvider } from '@sap-ux/system-access';
+import type { AbapServiceProvider, AxiosRequestConfig, ProviderConfiguration } from '@sap-ux/axios-extension';
+
+import type { EndpointsManager } from '../../common';
+
+export type RequestOptions = AxiosRequestConfig & Partial<ProviderConfiguration>;
+
+/**
+ * Service for managing and providing access to an ABAP service provider.
+ */
+export class AbapProvider {
+    private provider: AbapServiceProvider;
+    private isConnected: boolean;
+    private system: string | undefined;
+
+    /**
+     * Constructs an instance of AbapProvider.
+     *
+     * @param {EndpointsManager} endpointsManager - The endpoints service for retrieving system details.
+     * @param {ToolsLogger} [logger] - The logger.
+     */
+    constructor(private endpointsManager: EndpointsManager, private logger?: ToolsLogger) {
+        this.isConnected = false;
+        this.system = undefined;
+    }
+
+    /**
+     * Retrieves the configured ABAP service provider if set, otherwise throws an error.
+     *
+     * @returns {AbapServiceProvider} - The configured ABAP service provider.
+     */
+    public getProvider(): AbapServiceProvider {
+        if (!this.provider) {
+            throw new Error('Provider was not set!');
+        }
+        return this.provider;
+    }
+
+    /**
+     * Retrieves wheter the configured ABAP service provider is connected to ABAP system.
+     *
+     * @returns {boolean} - if provider is connected to ABAP system.
+     */
+    public getIsConnected(): boolean {
```

Comment:
why this wrapper? wouldn't be a getter/setter a nicer option if it is just about preventing it to be changed from the outside? If you really need this function then please rename it to `isConnected` and the variable to `connected`.

Reply:
Fixed in [d4795dc](https://github.com/SAP/open-ux-tools/pull/2169/commits/d4795dc11696b6184a7ec64bb12286b9bc077f06)